src/anim/evaluator/anim-target-value.js

```diff
@@ -28,22 +41,36 @@ class AnimTargetValue {
         this.valueType = type;
         this.dirty = true;
         this.value = [0, 0, 0, 1];
+        this.baseValue = null;
+        this.setter = null;
+    }
+
+    get _normalizeWeights() {
+        return this._component.normalizeWeights;
     }
 
     getWeight(index) {
         if (this.dirty) this.updateWeights();
-        if (this.totalWeight === 0 || !this.mask[index]) {
+        if (this._normalizeWeights && this.totalWeight === 0 || !this.mask[index]) {
             return 0;
+        } else if (this._normalizeWeights) {
+            return this.weights[index] / this.totalWeight;
         }
-        return this.weights[index] / this.totalWeight;
+        return math.clamp(this.weights[index], 0, 1);
+    }
+
+    _layerBlendType(index) {
+        return this._component.layers[index].blendType;
     }
 
     setMask(index, value) {
         this.mask[index] = value;
-        if (this._component.layers[index].blendType === ANIM_LAYER_OVERWRITE) {
-            this.mask = this.mask.fill(0, 0, index);
+        if (this._normalizeWeights) {
+            if (this._component.layers[index].blendType === ANIM_LAYER_OVERWRITE) {
+                this.mask = this.mask.fill(0, 0, index);
```

Comment:
this should probably execute in !normalizeWeights case a well, if the weight of the layer is 1?

Reply:
The mask is used by the normalised weight algorithm to removed these layers from the equation independent of their current weight value. Though I've added a check in the updateValue function to check whether a layer weight is 0 though, in which case it is also skipped.

Reply:
I understand. But what I was wondering is if there is a case where some layers should be 'ignored' in case normalizeWeights is false. Is there some setup in this case, where all underlaying layers are ignored? For example if the layer is 'overwrite' type (not additive) and its weight is 1? Or is there nothing like that when normalizeWeights is false.

Reply:
Nope the mask setup is only used for the normalizeWeights algorithm, theres nothing like that used when normalizeWeights is false.